### ISSUE TITLE
FormattedNumber doesn't take locale into account in tests

### ISSUE THREAD

**Issue by lauterry** [yahoo/react-intl#1112]
### Is this a BUG REPORT or FEATURE REQUEST? (choose one)

Bug report

### Expected behavior
With "fr" locale provided to IntlProvider, I expected the FormattedNumber to output "8 278 €"

### Current behavior
FormattedNumber outputs "€8,228"

### Step to reproduce for BUG REPORT

```
import React from 'react';
import { IntlProvider, intlShape } from 'react-intl';
import { mount } from 'enzyme';

const messages = require('../src/intl/messages/fr.json');

const intlProvider = new IntlProvider({locale: 'fr', messages}, {});
const {intl} = intlProvider.getChildContext();

function nodeWithIntlProp(node) {
	return React.cloneElement(node, {intl});
}

export function mountWithIntl(node, { context, childContextTypes } = {}) {
	return mount(
		nodeWithIntlProp(node),
		{
			context: Object.assign({}, context, {intl}),
			childContextTypes: Object.assign({}, { intl: intlShape }, childContextTypes)
		}
	);
}
```

```
	const wrapper = mountWithIntl(<FormattedNumber
			value={ 8278 }
			style="currency"
			currency={ "EUR" }
			currencyDisplay= "symbol"
			useGrouping= { true }
			minimumFractionDigits={ 0 }
			maximumFractionDigits={ 2 }>
			{(money) => (
				<span className="amount__value">{ money }</span>
			)}
		</FormattedNumber>);
		expect(wrapper.text()).toBe("8 278 €");
```

### Your Environment
Running test with Jest 22.4.2, enzyme 2.9.1, react-intl 2.4.0 on MacOS X 10.13.3

**Comment by papasmile** [comment]
@lauterry as you may know, actual rendering of final output is not in react-intl, but in Intl via Format.js [1].  Basically, rendering is eventually the responsibility of the browser.  But when running your test, you don't have one!

Specifically, your issues seems to stem from use of `text()`: "This function should be looked at with skepticism if being used to test what the actual HTML output of the component will be" [2]

When it comes to `render` (via Cheerio), it's also not designed for producing final input.  "Specifically, it does not produce a visual rendering, apply CSS, load external resources, or execute JavaScript. If your use case requires any of this functionality, you should consider projects like PhantomJS or JSDom." [3]

Long story short: you should not expect to validate exact text output from Intl in unit tests using Enzyme.  It's pretty safe to assume that Intl will work properly in any compliant browser environment. [4]

I hope that helps.

[1] https://formatjs.io/guides/runtime-environments/
[2] https://github.com/airbnb/enzyme/blob/master/docs/api/ReactWrapper/text.md
[3] https://cheerio.js.org/
[4] https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl

**Comment by lauterry** [comment]
@papasmile Thx to your answer and a little more investigation, I've been able to solve my problem following this [article](http://www.stevenatkin.com/index.php/2015/10/20/how-to-use-the-internationalization-apis-in-node-js-on-ibm-bluemix). 

I just : 

* ```npm install --save-dev full-icu```
* ```node --icu-data-dir=node_modules/full-icu node_modules/jest/bin/jest.js```

 and all FormattedDate and FormattedNumber outputs everything correctly according my fr locale
